spec/models/solidus_subscriptions/installment_spec.rb

```diff
@@ -168,71 +168,93 @@
     end
 
     context 'when maximum_reprocessing_attempts is configured' do
-      context 'when the installment has reached the maximum number of attempts' do
+      context 'when the installment has surpassed the maximum reprocessing time' do
         it 'creates a new installment detail' do
           allow(SolidusSubscriptions.configuration).to receive_messages(
-            maximum_reprocessing_attempts: 3,
+            maximum_reprocessing_time: 3.days,
             reprocessing_interval: 2.days,
           )
-          installment = create(:installment)
-          create_list(:installment_detail, 2, installment: installment, success: false)
-
-          installment.payment_failed!(create(:order))
+          subscription = create(:subscription)
+          _last_successful_installment = create(
```

Comment:
Is there a reason you're assigning these to a variable?

Reply:
I've done this to make it clear what was the meaning of the created object by also being explicit about the time it occurred.